### PR TITLE
feature(fetching data from variants endpoint) 

### PR DESCRIPTION
**Pull request by shravanipagilla**
# Description:
 Add dynamic “color”, “size” and “quantity” from DB in the product view(ByProductId)

# Jira:
https://hackyourfuture-dk.atlassian.net/jira/software/projects/CLASS22/boards/5?selectedIssue=CLASS22-107

### CODE REVIEW

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (shpomp) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: 🗑️

**Review by shpomp — CHANGES_REQUESTED** [review]
commented code

**Review by ShravyaGurmitkal — DISMISSED** [review]

**Review by shpomp — CHANGES_REQUESTED** [review]
The branch is "CLASS22-107-add-dynamic-variants-from-db-in-product-view", the PR name  "feature(fetching data from variants endpoint)", and you are NOT fetching any data from any endpoint here.

File: packages/client/src/components/ViewDropDown/ViewDropdown.component.js
Comment (miltonhultgren) [inline packages/client/src/components/ViewDropDown/ViewDropdown.component.js]: Please remove this and fix the prop types instead. You want to use `arrayOf(PropTypes.shape(...))`.

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (miltonhultgren) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: Why do these have capital names?

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (miltonhultgren) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: This should all be controlled by React `useState` so that it doesn't happen on each render.
Once you connect the API that might make more sense and consider using `Set`.

**Review by miltonhultgren — CHANGES_REQUESTED** [review]

**Comment by ShravyaGurmitkal** [comment]
@shpomp Yes we have first doane using fetching like below and commented the code 
  // const [variants, setVariants] = useState([]);

  // useEffect(() => {
  //   const fetchVariantsByProductId = async () => {
  //     const response = await fetch(`${apiURL()}/variants?product=${productId}`);
  //     const variantsProductData = await response.json();
  //     setVariants(variantsProductData);
  //   };

  //   fetchVariantsByProductId();
  // }, [variants]);

We thought to uncomment it when the endpoint is merged and make some changes as needed in this branch and commit those changes.
But we removed it as it is commented code as you said and pushed the branch.

File: packages/client/src/components/ViewDropDown/ViewDropdown.component.js
Comment (ShravyaGurmitkal) [reply packages/client/src/components/ViewDropDown/ViewDropdown.component.js]: @miltonhultgren we have tried with arrayOf(PropTypes.shape(...)) but we are getting error as we have both strings and numbers so we have changed it to .any . Please suggest us how to use that and change it to arrayOf(PropTypes.shape(...)). Thanks

**Comment by shpomp** [comment]
> 

Now I understand, thanks. 
The point is that there is 0 information about it anywhere, somebody has to spend time trying to understand, then ask, then wait for a reply, etc. 
It is your responsibility to make it clear and it is very easy to do with simple descriptions and comments before somebody has to get confused and ask. 🙏🏽

**Review by shpomp — DISMISSED** [review]

**Comment by shpomp** [comment]
❗ I approved in order to unblock my change request, but you should address all that Milton said. 
@ShravyaGurmitkal

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (ShravyaGurmitkal) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: I will change them

**Comment by shravanipagilla** [comment]
> @shpomp Yes we have first doane using fetching like below and commented the code // const [variants, setVariants] = useState([]);
> 
> // useEffect(() => { // const fetchVariantsByProductId = async () => { // const response = await fetch(`${apiURL()}/variants?product=${productId}`); // const variantsProductData = await response.json(); // setVariants(variantsProductData); // };
> 
> // fetchVariantsByProductId(); // }, [variants]);
> 
> We thought to uncomment it when the endpoint is merged and make some changes as needed in this branch and commit those changes. But we removed it as it is commented code as you said and pushed the branch.

Thank you shravya for explaining to @shpomp about our code .... yes @shpomp  as she said we did that one but I was in traveling so I did not reply to you .. sorry about it...

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (shravanipagilla) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: Hej @miltonhultgren it doesn't have any meaning to write.. but I did usual, and we made changes... thank you ..

**Comment by shravanipagilla** [comment]
> > 
> 
> Now I understand, thanks. The point is that there is 0 information about it anywhere, somebody has to spend time trying to understand, then ask, then wait for a reply, etc. It is your responsibility to make it clear and it is very easy to do with simple descriptions and comments before somebody has to get confused and ask. 🙏🏽

@shpomp I was planning to write to you, but I was very busy for 2 days and also traveling. sorry about it I will do my best next time.. Thank you ...

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (shpomp) [inline packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: I would go back to Milton's suggestion here to use Set. 
There were 2 thumbs up on that comment, but no reply and no change either. Sneaky! 😅 


image

**Review by shpomp — CHANGES_REQUESTED** [review]

File: packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js
Comment (ShravyaGurmitkal) [reply packages/client/src/containers/DropdownGroup/DropdownGroup.Container.js]: I thought milton commented as I didn't use usestate for variants , after fetching I have used usestate for variants. sorry, will change now

**Review by shpomp — APPROVED** [review]

**Comment by miltonhultgren** [comment]
Since Karo already approved this I won't look again :)

**Review by miltonhultgren — APPROVED** [review]

**Pull request merged**
